lib/itk/queue/connection.ex

```diff
@@ -61,33 +85,29 @@ defmodule ITKQueue.Connection do
     [host: host, port: port, username: username, password: password, heartbeat: heartbeat]
   end
 
+  # This will block the GenServer until connection is opened successfully.
   defp connect(params) do
-    AMQP.Connection.open(params)
+    params
+    |> AMQP.Connection.open()
     |> handle_connection_result(params)
   end
 
-  defp handle_connection_result({:ok, connection}, _params) do
-    Process.link(connection.pid)
-    connection
+  defp handle_connection_result({:ok, connection}, params) do
+    Process.monitor(connection.pid)
+    %__MODULE__{params: params, connection: connection}
   end
 
   defp handle_connection_result({:error, _}, params) do
     Logger.info("AMQP connection failed, retrying")
-    Process.sleep(1000)
+    Process.sleep(@reconnect_interval)
     connect(params)
   end
 
   @doc """
-    handles server down
+  Handles termination of this GenServer.
   """
-  def terminate(reason, connection) do
-    Logger.info("Terminating AMQP connection: #{inspect(reason)}")
-
-    if Process.alive?(connection.pid) do
-      Logger.info("Closing AMQP connection")
-      AMQP.Connection.close(connection)
-    end
-
+  def terminate(reason, _state) do
+    Logger.info("Terminating AMQP connection manager: #{inspect(reason)}")
```

Comment:
Can you elaborate more on why we're not closing the connection manually here?

Reply:
Sorry I left that part out, I'll add it now, it can happen when the overflow workers are killed.